ops/pebble.py

```diff
@@ -1135,6 +1135,18 @@ def __eq__(self, other: Union['CheckDict', 'Check']) -> bool:
         else:
             return NotImplemented
 
+    def _merge(self, other: 'Check'):
+        """Merges this check object with another check definition.
+
+        For attributes present in both objects, the passed in check
+        attributes take precedence.
+        """
+        for name, value in other.__dict__.items():
+            if not value or name == 'name':
+                continue
+            else:
```

Comment:
We need to consider the "http", "tcp", and "exec" fields and merge those too. Something like this:

```python
if not value or name == 'name':
    continue
if name == 'http':
    self._merge_http(value)
elif name == 'tcp':
    self._merge_tcp(value)
elif name == 'exec':
    self._merge_exec(value)
else:
    setattr(self, name, value)

def _merge_http(self, other_dict):
    # handle "headers" specially, kind of like:
    # https://github.com/canonical/pebble/blob/d822e9fd9c484a9c35e1b8f08b8d438ca2115225/internals/plan/plan.go#L398

...
"""

Reply:
Would you be willing to do the same for log-targets while you're in here? See also https://github.com/canonical/operator/pull/1194, which tried to do this, but got stale. The original PR owner hasn't responded for a long time, but -- speaking of "merging", maybe we could merge these efforts?

Reply:
I added the _merge methods now and I'll start working on the log-targets, they look very similar to this 

BTW I ended up doing some mess while rebasing this branch so other commits are appearing right now. Is that ok? I believe once the PR is approved, will be squashed right?

Reply:
Done :)

Reply:
ah one more thing, about header wouldnt be enough doing the `getattr(self.http, name).update(value)` since is a ` Dict[str, str]`?